src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionResultSimulationService.java

```diff
@@ -0,0 +1,178 @@
+        if (optionalStudentParticipation.isEmpty()) {
+            programmingExerciseStudentParticipation = createParticipation(programmingExercise, participant, user);
+        }
+        else {
+            programmingExerciseStudentParticipation = (ProgrammingExerciseStudentParticipation) optionalStudentParticipation.get();
+        }
+
+        ProgrammingSubmission programmingSubmission = new ProgrammingSubmission();
+        programmingSubmission.setCommitHash(VCSSimulationUtils.simulateCommitHash());
+        programmingSubmission.setSubmitted(true);
+        programmingSubmission.setSubmissionDate(ZonedDateTime.now());
+        programmingSubmission.setType(SubmissionType.MANUAL);
+        programmingExerciseStudentParticipation.addSubmissions(programmingSubmission);
+
+        programmingSubmissionRepository.save(programmingSubmission);
+        return programmingSubmission;
+    }
+
+    /**
+     *  This method creates a new result for the provided participation
+     *  This functionality is only for testing purposes (noVersionControlAndContinuousIntegrationAvailable)
+     * @param programmingExerciseStudentParticipation the participation for which the new result should be created
+     * @return the newly created and stored result
+     */
+    public Result createResult(ProgrammingExerciseStudentParticipation programmingExerciseStudentParticipation) {
+        Optional<ProgrammingSubmission> programmingSubmission = programmingSubmissionRepository
+                .findFirstByParticipationIdOrderBySubmissionDateDesc(programmingExerciseStudentParticipation.getId());
+        Result result = new Result();
+        result.setSubmission(programmingSubmission.get());
+        result.setParticipation(programmingExerciseStudentParticipation);
+        result.setRated(true);
+        result.resultString("7 of 13 passed");
+        result.score(54L);
+        result.setAssessmentType(AssessmentType.AUTOMATIC);
+        result.setCompletionDate(ZonedDateTime.now());
+        this.addFeedback(result);
+        resultRepository.save(result);
+        return result;
+    }
+
+    /**
+     * Creates feedback for the provided result
+     * This functionality is only for testing purposes (noVersionControlAndContinuousIntegrationAvailable)
+     * @param result for which the feedback should be created
+     * @param methodName of the testcase
+     * @param positive is the testcase positive or not
+     * @param errorMessageString will only added if the test case fails otherwise use null
+     */
+    public void createFeedback(Result result, String methodName, boolean positive, @Nullable String errorMessageString) {
+        Feedback feedback = new Feedback();
+        feedback.setText(methodName);
+        feedback.setDetailText(errorMessageString);
+        feedback.setType(FeedbackType.AUTOMATIC);
+        feedback.setPositive(positive);
+        result.addFeedback(feedback);
+    }
+
+    /**
+     * adds the feedback to the result
```

Comment:
please add something like "currently only the Java standard template is supported"

Reply:
i added the comment